### PR TITLE
Code Quality: Reverting changes for DateTime formatting without ICU

### LINKED ISSUE #15554: Bug: Files slows to a crawl if opening a folder with lots of files (preview build)
### Description

Effectively if I open a folder that has lots of files with varying sizes (between 1kb-9GB), and the folder contains around 2000+ Files will just slow to a crawl until I close the tab with said folder.

### Steps To Reproduce

1. Open files
2. Navigate to the folder with 2000+ items
3. It'll slow to a crawl after a few seconds

### Requirements

To open the folder with minimum delay between reading/loading all the files in a folder with large amount of files.

### Files Version

3.4.5

### Windows Version

10.0.19045.4170

### Log File

[debug.log](https://github.com/user-attachments/files/15571927/debug.log)

### PR DESCRIPTION
**Pull request by XTorLukas**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #15554

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Try loading bigger folder e.g. `C:/Windows/System32`
2. It is now faster to load strings with formatting using `GetLocalizedFormatResource`

### CODE REVIEW

**Comment by XTorLukas** [comment]
@yaira2 fyi

File: src/Files.App/Services/DateTimeFormatter/ApplicationDateTimeFormatter.cs [hunk 1]
```diff
@@ -1,48 +1,80 @@
...
 // Licensed under the MIT License. See the LICENSE.
 
-using Files.App.Extensions;
-using System;
+using System.Text;
 
 namespace Files.App.Services.DateTimeFormatter
 {
+	/// <summary>
+	/// Application-specific implementation of the DateTimeFormatter.
```
Comment (0x5bfa) [inline src/Files.App/Services/DateTimeFormatter/ApplicationDateTimeFormatter.cs:9]:
```suggestion
	/// Represents service for application-specific <see cref=“DateTimeOffset”/> formatter.
```

**Review by 0x5bfa — APPROVED** [review]
Those are used in bulk enumeration so speed is important imo.
Do you think having our own message formatter *would* improve performance?

Codewise, looks good to me.

**Comment by XTorLukas** [comment]
> Those are used in bulk enumeration so speed is important imo. Do you think having our own message formatter _would_ improve performance?

According to the benchmarks I am doing while developing my own UnicodeFormatter, the results are promising so far, but I am still working on optimisation and development, here I am already running into the limits of C# and it is necessary to redirect some operations to the C++ part or use pointers in the unsafe part of the code. I will continue the development. For now, however, this MessageFormat cannot be used with a large number of elements.

**Review by yaira2 — APPROVED** [review]
Hopefully we can figure this out so that we can bring back the code improvements, but the main thing is that performance is back to how it was before.

**Pull request merged**